include/boost/geometry/algorithms/detail/calculate_point_order.hpp

```diff
@@ -36,7 +36,7 @@ struct clean_point
         , m_is_azi_valid(false), m_is_azi_diff_valid(false)
     {}
 
-    typename boost::iterators::iterator_reference<Iter>::type ref() const
+    auto ref() const
```

Comment:
This way the reference will be converted to value. You should probably return `decltype(auto)`.

Reply:
:heavy_check_mark: reverted

Reply:
FYI, revert was not required. Returning `decltype(auto)` would be enough.

Reply:
:heavy_check_mark: intuitively it looked the same, but I get it now. Changed. Thank you.